src/Http/FlashMessage.php

```diff
@@ -0,0 +1,195 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         4.2.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Http;
+
+use Cake\Core\InstanceConfigTrait;
+use Cake\Utility\Hash;
+use Throwable;
+
+/**
+ * The FlashMessage class provides a way for you to write a flash variable
+ * to the session, to be rendered in a view with the FlashHelper.
+ */
+class FlashMessage
```

Comment:
Should the request have a method or an attribute that exposes flash messages? Or perhaps a method on the `Session` object?

```php
// My preference, but doesn't follow the `getXYZ` pattern.
$this->request->flash()->set(...);

// Simple & mirrors how sessions are exposed.
$this->request->getFlash()->set(...);

// Fits within the PSR spec, and we could still have getFlash() or flash()
$this->request->getAttribute('flash')->set(...);

// Is a bit more verbose, but workable.
$this->request->getSession()->getFlash()->set(...);
```

Reply:
Since there's no corresponding setter method I would prefer `request->flash()`, (even though for session we have `getSession()`). We deprecated `Behavior::setTable()` in favor of `Behavior::table()` for the same reason.

Internally it would be set as a request attribute so `request->getAttribute('flash')` would also work.

Reply:
I think behavior table() was because of collision with Table::setTable() (meaning table string of database table)
But I agree, internally as attribute might suffice for the 4.x lifecycle
And with more experience on the subject we can still think about changes for 5.x when the time comes.

Reply:
> I think behavior table() was because of collision with Table::setTable() (meaning table string of database table)

Ah, I forgot that. Well I can live with `request->getFlash()` too, `request->getAttribute('flash')` too cumbersome and can't be auto completed by IDEs.

Reply:
It can with IdeHelper help for PHPStorm at least ;)

But all right, getFlash() sounds reasonable then.